<?xml version="1.0"?>

<launch>
    <arg name="bag_out_filename"    value="syn_meas"/>
    <arg name="boat_imu"            value="/base_imu"/>
    <arg name="base_2_rover_relPos" value="/base2Rover_relPos"/>
    <arg name="rover_PosVelEcef"    value="/rover_gps"/>
    <arg name="boat_PosVelEcef"     value="/base_gps"/>
    <arg name="compass_relPos"      value="/base_rtk_compass"/>
    <arg name="ref_lla"             value="/ref_lla"/>
    <arg name="params_file"         value="params"/>

    <include file="$(find boat_estimator)/launch/standard.launch">
        <arg name="boat_imu"            value="$(arg boat_imu)"/>
        <arg name="base_2_rover_relPos" value="$(arg base_2_rover_relPos)"/>
        <arg name="rover_PosVelEcef"    value="$(arg rover_PosVelEcef)"/>
        <arg name="boat_PosVelEcef"     value="$(arg boat_PosVelEcef)"/>
        <arg name="compass_relPos"      value="$(arg compass_relPos)"/>
        <arg name="params_file"         value="$(arg params_file)"/>
    </include>

    <node name="syn_meas" pkg="boat_estimator" type="syn_meas_class.py" output="screen"/>

    <node pkg="rosbag" type="record" name="record" output="screen" args="-O ../data/px4flight/sim/$(arg bag_out_filename).bag --all"/>
</launch>
